File: Spherical Adventures/Assets/Scripts/Teleport.js
#pragma strict
var target : GameObject;
public var TeleportEffect : Transform;
public var TeleportEffect2 : Transform;
var maxFallDistance = -175;
var lvl1 = "Level1";


function Update () 
{
	if (Input.GetKeyDown (KeyCode.R)){
		if (GameMaster.teleports <= 0)
		{
			Debug.Log("No more Teleports Left!");
		}
		else 
		{
			GameMaster.teleports -= 1;
			gameObject.transform.position = Vector3(target.transform.position.x, target.transform.position.y, target.transform.rotation.x);
			var effect = Instantiate(TeleportEffect, transform.position, transform.rotation);
			transform.Rotate(Vector3.right * Time.deltaTime);
			Destroy(effect.gameObject, 1.1);
		}	
	}
	if (transform.position.y <= maxFallDistance)
	{
		if (GameMaster.teleports <= 0)
		{
			Application.LoadLevel(lvl1);
		}
		else 
		{
			GameMaster.teleports -= 1;
			gameObject.transform.position = Vector3(target.transform.position.x, target.transform.position.y, target.transform.rotation.x);
			var effect2 = Instantiate(TeleportEffect, transform.position, transform.rotation);
			transform.Rotate(Vector3.right * Time.deltaTime);
			Destroy(effect2.gameObject, 1.1);
		}
	}
}
function OnCollisionEnter(col : Collision) {
     if (col.collider.name == "Barrel" ) {
         	GameMaster.teleports -= 1;
			gameObject.transform.position = Vector3(target.transform.position.x, target.transform.position.y, target.transform.rotation.x);
			var effect3 = Instantiate(TeleportEffect, transform.position, transform.rotation);
			transform.Rotate(Vector3.right * Time.deltaTime);
			Destroy(effect3.gameObject, 1.1);
     }
 }

File: Assets/Scripts/Teleport.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class Teleport : MonoBehaviour
{
    public GameObject target;
    public Transform TeleportEffect;
    public Transform TeleportEffect2;
    public int maxFallDistance;
    public string lvl1;
    public virtual void Update()
    {
        if (Input.GetKeyDown(KeyCode.R))
        {
            if (GameMaster.teleports <= 0)
            {
                Debug.Log("No more Teleports Left!");
            }
            else
            {
                GameMaster.teleports = GameMaster.teleports - 1;
                this.gameObject.transform.position = new Vector3(this.target.transform.position.x, this.target.transform.position.y, this.target.transform.rotation.x);
                Transform effect = UnityEngine.Object.Instantiate(this.TeleportEffect, this.transform.position, this.transform.rotation);
                this.transform.Rotate(Vector3.right * Time.deltaTime);
                UnityEngine.Object.Destroy(effect.gameObject, 1.1f);
            }
        }
        if (this.transform.position.y <= this.maxFallDistance)
        {
            if (GameMaster.teleports <= 0)
            {
                Application.LoadLevel(this.lvl1);
            }
            else
            {
                GameMaster.teleports = GameMaster.teleports - 1;
                this.gameObject.transform.position = new Vector3(this.target.transform.position.x, this.target.transform.position.y, this.target.transform.rotation.x);
                Transform effect2 = UnityEngine.Object.Instantiate(this.TeleportEffect, this.transform.position, this.transform.rotation);
                this.transform.Rotate(Vector3.right * Time.deltaTime);
                UnityEngine.Object.Destroy(effect2.gameObject, 1.1f);
            }
        }
    }

    public virtual void OnCollisionEnter(Collision col)
    {
        if (col.collider.name == "Barrel")
        {
            GameMaster.teleports = GameMaster.teleports - 1;
            this.gameObject.transform.position = new Vector3(this.target.transform.position.x, this.target.transform.position.y, this.target.transform.rotation.x);
            Transform effect3 = UnityEngine.Object.Instantiate(this.TeleportEffect, this.transform.position, this.transform.rotation);
            this.transform.Rotate(Vector3.right * Time.deltaTime);
            UnityEngine.Object.Destroy(effect3.gameObject, 1.1f);
        }
    }

    public Teleport()
    {
        this.maxFallDistance = -175;
        this.lvl1 = "Level1";
    }

}